webapp/src/main/java/life/qbic/usermanagement/passwordreset/PasswordResetLinkSupplier.java

```diff
@@ -11,28 +13,32 @@
  */
 class PasswordResetLinkSupplier {
 
+  private final String protocol;
+
   private final String host;
 
-  private final String port;
+  private int port;
 
-  private final String loginEndpoint;
+  private final String resetEndpoint;
 
-  private final String emailConfirmationParameter;
+  private final String passwordResetParameter;
 
-  public PasswordResetLinkSupplier(@Value("${host.name}") String host,
-      @Value("${server.port}") String port,
-      @Value("${login-endpoint}") String loginEndpoint,
-      @Value("${email-password-reset-parameter}") String emailConfirmationParameter) {
+  public PasswordResetLinkSupplier(
+      @Value("${service.host.protocol}") String protocol,
+      @Value("${service.host.name}") String host,
+      @Value("${server.port}") int port,
+      @Value("${password-reset-endpoint}") String resetEndpoint,
+      @Value("${email-password-reset-parameter}") String passwordResetParameter) {
+    this.protocol = protocol;
     this.host = host;
     this.port = port;
-    this.loginEndpoint = loginEndpoint;
-    this.emailConfirmationParameter = emailConfirmationParameter;
+    this.resetEndpoint = resetEndpoint;
+    this.passwordResetParameter = passwordResetParameter;
   }
 
-  public String passwordResetUrl(String userId) {
-    String hostAddress = String.join(":", host, port);
-    String params = String.join("=", emailConfirmationParameter, userId);
-    String endpointWithParams = String.join("?", loginEndpoint, params);
-    return String.join("/", hostAddress, endpointWithParams);
+  public String passwordResetUrl(String userId) throws MalformedURLException {
```

Comment:
I love that you included a MalformedURLException! 🧠
I'd argue that this should also be tested in the Spec(e.g. "htt" as protocol)

Reply:
Not needed, that is part of the java.net.URL and it will work ;)